src/lib.rs

```diff
@@ -958,7 +958,14 @@ pub trait Function {
     /// match -- specifically, the returned metadata may cover only a
     /// subset of the requested ranges -- if the value is not live for
     /// the entire requested ranges.
-    fn debug_value_labels(&self) -> &[(Inst, Inst, VReg, u32)] {
+    ///
+    /// The instruction indices imply a program point just *before*
+    /// the instruction.
+    ///
+    /// Preconditions: we require this slice to be sorted in
+    /// lexicographic order (i.e., first by vreg, then by instruction
+    /// index), and we require the ranges to be non-overlapping.
+    fn debug_value_labels(&self) -> &[(VReg, Inst, Inst, u32)] {
```

Comment:
I'm having trouble seeing how passing a range along with a `VReg` is useful here. Why would a single `VReg` have more than one debug label in practice?

A much simpler API would be:
```rust
fn debug_value_label(&self, vreg: VReg) -> Option<u32> { ... }
```

Reply:
Yeah, I agree that would be a lot simpler. Unfortunately at least the debug support in Cranelift allows for labels to be attached to `Value`s only for certain ranges, so we need to carry this through. I believe that this comes up for example when a Wasm-level local is represented by multiple SSA values (due to control flow and multiple assignments); any given label-to-vreg mapping needs bounds that are carried through to the DWARF. And, the `cranelift-frontend` API has `FunctionBuilder::set_val_label` which allows attaching an arbitrary number of labels to a value, starting at arbitrary program points. We could simplify that first, but IMHO I'd rather allow for the more general approach (especially if relatively low-cost) at the regalloc API level; especially when we can't know what other users might require.